fog/view/server/test-utils/src/lib.rs

```diff
@@ -369,3 +443,71 @@ pub fn wait_for_server_to_load(
         sleep(Duration::from_secs(1));
     }
 }
+
+/// Wait until a server has added a specific number of blocks to load.
+pub fn wait_for_block_to_load(block_count: u64, store_servers: &[TestViewServer], logger: &Logger) {
+    let mut allowed_tries = 60usize;
+    loop {
+        let server_num_blocks = get_highest_processed_block_count(store_servers);
+        if server_num_blocks >= block_count {
+            break;
+        }
+        log::info!(
+            logger,
+            "Waiting for server to catch up to db... {} < {}",
+            server_num_blocks,
+            block_count,
+        );
+        if allowed_tries == 0 {
+            panic!("Server did not catch up to database!");
+        }
+        allowed_tries -= 1;
+        sleep(Duration::from_millis(1000));
+    }
+}
+
+/// Wait until a server has added a specific number of blocks to load.
+pub fn wait_for_highest_processed_and_last_known(
+    view_client: &mut FogViewGrpcClient,
+    highest_processed_block_count: u64,
+    last_known_block_count: u64,
+) {
+    let mut allowed_tries = 60usize;
+    loop {
+        let nonsense_search_keys = vec![vec![50u8]];
+        let result = view_client.request(0, 0, nonsense_search_keys).unwrap();
+        if result.highest_processed_block_count == highest_processed_block_count
+            && result.last_known_block_count == last_known_block_count
```

Comment:
I think we should keep it because the way it's used in tests is to make sure the proper block counts are returned (not more). Moreover, hesitant to change this test logic because I want to make it as similar to the existing unary tests as possible. I think it's somewhat risky to change the test without much upside.